public_html/storefront/controller/pages/account/download.php

```diff
@@ -76,12 +76,11 @@ public function main()
         );
 
         if (isset($this->request->get['limit'])) {
-            $limit = (int) $this->request->get['limit'];
-            $limit = $limit > 50 ? 50 : $limit;
-        } else {
-            $limit = $this->config->get('config_catalog_limit');
+           $limit = (int) $this->request->get['limit'];
```

Comment:
in case when get['limit'] presents you should to set some maximum because performance issue. When somebody send 100000 as limit parameter your database will goes down.

Try this:
if ( (int)$this->request->get['limit'] ) {
           $limit = min((int) $this->request->get['limit'], 50);
}

Reply:
Thanks for the review abolabo. i have changed it.

Please check again.